geomstats/geometry/functions.py

```diff
@@ -0,0 +1,245 @@
+        )
+
+        return out
+
+
+class SinfSpace(Manifold):
+    """Class for space of L2 functions with norm 1.
+
+    The tangent space is given by functions that have
+    zero inner-product with the base point
+
+    Parameters:
+    -------
+    domain_samples : array of shape (n_samples, )
+        grid points on the domain
+
+    Ref :
+    -----
+    .. [Sea2016] Srivastava, Anuj, and Eric P. Klassen.
+    Functional and shape data analysis.
+    Vol. 1. New York: Springer, 2016.
+    """
+
+    def __init__(self, domain_samples):
+        self.domain = domain_samples
+        super().__init__(dim=math.inf, metric=SinfSpaceMetric(self.domain))
+
+    def projection(self, point):
+        """Project a point to the infinite dimensional hypersphere.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., dim]
+            discrete evaluation of a function.
+        Returns
+        -------
+        projected_point : array-like, shape=[..., dim]
+            Point projected to the hypersphere.
+        """
+        norm_p = self.metric.norm(point)
+        return point / norm_p
+
+    def belongs(self, point, atol=gs.atol):
+        """Evaluate if the point belongs to the vector space.
+
+        This method checks the shape of the input point.
+        Parameters
+        ----------
+        point : array-like, shape=[.., {dim, [n, n]}]
+            Point to test.
+        atol : float
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the space.
+        """
+        norms = self.metric.norm(point)
+
+        return gs.isclose(norms - 1.0, atol)
+
+    def is_tangent(self, vector, base_point, atol=gs.atol):
+        """Check whether the vector is tangent at base_point.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., dim]
+            Vector.
+        base_point : array-like, shape=[..., dim]
+            Point on the manifold.
+        atol : float
+            Absolute tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        is_tangent : bool
+            Boolean denoting if vector is a tangent vector at the base point.
+        """
+        inner_product = self.metric.inner_product(vector, base_point)
+
+        return gs.isclose(inner_product, atol)
+
+    def to_tangent(self, vector, base_point):
+        """Project a vector to a tangent space of the manifold.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., dim]
+            Vector.
+        base_point : array-like, shape=[..., dim]
+            Point on the manifold.
+
+        Returns
+        -------
+        tangent_vec : array-like, shape=[..., dim]
+            Tangent vector at base point.
+        """
+        inner_product = self.metric.inner_product(vector, base_point)
+        tangent_vec = vector - inner_product
```

Comment:
Is this right? inner_product is a scalar, shouldn't it be multiplied by a base_point?

Reply:
Yeah, that was wrong. I updated this code to do this properly. Thanks for pointing this out again.